### ISSUE TITLE
Multiline property wrappers cause false positives for `let_var_whitespace`

### ISSUE THREAD

**Issue by mildm8nnered** [realm/SwiftLint#4801]
### New Issue Checklist

- [X] Updated SwiftLint to the latest version
- [X] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

Given the following code, `let_var_whitespace` will fire at lines 4 and 8.

```
internal struct Foo {
    @Flag(name: [.long, .customLong("autocorrect")], help: "Correct violations whenever possible.")
    internal var fix = false
    @Flag(help: """
        Should reformat the Swift files using the same mechanism used by Xcode (via SourceKit).
        Only applied with `--fix`/`--autocorrect`.
        """)
    internal var format = false
    @Flag(help: "Use an alternative algorithm to exclude paths for `excluded`, which may be faster in some cases.")
    internal var useAlternativeExcluding = false
}
```

This code is actually from LintOrAnalyzeArguments.swift

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
% swiftlint lint --no-cache --enable-all-rules Foo.swift
Linting Swift files at paths Foo.swift
Linting 'Foo.swift' (1/1)
The `anyobject_protocol` rule is now deprecated and will be completely removed in a future release.
/tmp/Foo.swift:3:14: warning: Explicit Type Interface Violation: Properties should have a type interface (explicit_type_interface)
/tmp/Foo.swift:8:14: warning: Explicit Type Interface Violation: Properties should have a type interface (explicit_type_interface)
/tmp/Foo.swift:10:14: warning: Explicit Type Interface Violation: Properties should have a type interface (explicit_type_interface)
/tmp/Foo.swift:4:5: warning: Variable Declaration Whitespace Violation: Let and var should be separated from other statements by a blank line. (let_var_whitespace)
/tmp/Foo.swift:8:5: warning: Variable Declaration Whitespace Violation: Let and var should be separated from other statements by a blank line. (let_var_whitespace)
Done linting! Found 5 violations, 0 serious in 1 file.
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? 0.50.3
* Installation method used (Homebrew, CocoaPods, building from source, etc)? Homebrew
* Paste your configuration file:

none

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?
  If so, paste their relative paths and respective contents.

no

* Which Xcode version are you using (check `xcodebuild -version`)?

```
Xcode 14.2
Build version 14C18
```

* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`
  to quickly test if your example is really demonstrating the issue. If your example is more
  complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.

See above.

**Comment by florentdouine** [comment]
Hi !
I did the same observation on another kind of code, if it could help the resolution : 
```swift
struct Example {
    @available(
        *, deprecated, renamed: "buttonStyle", message: """
    XXX
    """
    )
    public var style: Style = .primary { /// <-- Swiftlint warning `let_var_whitespace` here
        didSet {
            self.buttonStyle = .init(style: self.style, colorScheme: self.colorScheme)
        }
    }
}
